jetbrains-core/src/software/aws/toolkits/jetbrains/utils/ui/UiUtils.kt

```diff
@@ -21,14 +21,26 @@ import javax.swing.JComponent
 import javax.swing.JTextField
 import javax.swing.ListModel
 
-fun <T> ComboBox<T>.populateValues(selected: T? = null, updateStatus: Boolean = true, block: () -> Collection<T>) {
+/**
+ * @param selected The default selected item
+ * @param updateStatus If enabled, update the combo box not editable if the items are empty
+ * @param usePreviouslySelected If enabled, use previously selected item if it not not null instead of [selected]
+ * @param block Lambda function that returns a new set of items for the combo box.
+ */
+fun <T> ComboBox<T>.populateValues(
+    selected: T? = null,
+    updateStatus: Boolean = true,
+    usePreviouslySelected: Boolean = false,
+    block: () -> Collection<T>
+) {
     ApplicationManager.getApplication().executeOnPooledThread {
         val values = block()
         ApplicationManager.getApplication().invokeLater({
             val model = this.model as DefaultComboBoxModel<T>
+            val previouslySelected = model.selectedItem
             model.removeAllElements()
             values.forEach { model.addElement(it) }
-            this.selectedItem = selected
+            this.selectedItem = if (usePreviouslySelected && previouslySelected != null) previouslySelected else selected
```

Comment:
The trick is that the population code is an async call. This means when you are calling `view.runtime.populateValues`, the `view.runtime.selectedItem` is still `null`, while when you executing the code here, the `selectedItem` is forcefully set from loading the xml file even it is not populated.

This change is to avoid the overriding the default select from the settings.